optional field in optional struct gives C failure

### Describe the bug

The code below is crashing. Having an optional argument in a struct that is optional is not compiling. 
```v
module main

import json 

pub struct SomeStruct{
pub mut:
	test ?string
}

pub struct MyStruct{
pub mut:
	result ?SomeStruct
	id string
}

fn main() {
	a := MyStruct {
		id: "some id"
		result: SomeStruct{

		}
	}

	encoded_string := json.encode(a)
	println(encoded_string)
	test := json.decode(MyStruct, encoded_string) or {
		eprintln(err)
		exit(1)
	}
	println("All ok")
}
```

### Expected Behavior

See description

### Current Behavior
```
[jalon@7950x v]$ v -g run foo.v
foo.v:26:2: warning: unused variable: `test`
   24 |     encoded_string := json.encode(a)
   25 |     println(encoded_string)
   26 |     test := json.decode(MyStruct, encoded_string) or {
      |     ~~~~
   27 |         eprintln(err)
   28 |         exit(1)
/tmp/v_1000/foo.14030647679007329693.tmp.c:2327: warning: assignment from incompatible pointer type
/tmp/v_1000/foo.14030647679007329693.tmp.c:2335: error: field not found: test
builder error: 
==================
C error. This should never happen.
...
```
### Reproduction Steps

See description

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.3 f41ca48

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 5454562.f41ca48
OS: linux, Ubuntu 22.04.2 LTS
Processor: 16 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz

getwd: /work/web3_proxy/scripts
vexe: /work/v/v
vexe mtime: 2023-04-12 07:50:00

vroot: OK, value: /work/v
VMODULES: OK, value: /home/brandon/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.34.1
Git vroot status: weekly.2023.13-93-gf41ca48a
.git/config present: true

CC version: cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

--- a/vlib/json/json_option_struct_test.v
+++ b/vlib/json/json_option_struct_test.v
@@ -1,0 +1,23 @@
+import json
+
+pub struct SomeStruct {
+pub mut:
+	test ?string
+}
+
+pub struct MyStruct {
+pub mut:
+	result ?SomeStruct
+	id     string
+}
+
+fn test_main() {
+	a := MyStruct{
+		id: 'some id'
+		result: SomeStruct{}
+	}
+	encoded_string := json.encode(a)
+	assert encoded_string == '{"result":{},"id":"some id"}'
+	test := json.decode(MyStruct, encoded_string)!
+	assert test == a
+}

--- a/vlib/v/gen/c/json.v
+++ b/vlib/v/gen/c/json.v
@@ -673,15 +673,15 @@ fn (mut g Gen) gen_struct_enc_dec(utyp ast.Type, type_info ast.TypeInfo, styp st
 		is_option := field.typ.has_flag(.option)
 		indent := if is_option { '\t\t' } else { '\t' }
 		if is_option {
-			enc.writeln('\tif (val${op}${c_name(field.name)}.state != 2) {')
+			enc.writeln('\tif (${prefix_enc}${op}${c_name(field.name)}.state != 2) {')
 		}
 		if is_omit_empty {
 			if field.typ.has_flag(.option) {
-				enc.writeln('${indent}if (val${op}${c_name(field.name)}.state != 2)')
+				enc.writeln('${indent}if (${prefix_enc}${op}${c_name(field.name)}.state != 2)')
 			} else if field.typ == ast.string_type {
-				enc.writeln('${indent}if (val${op}${c_name(field.name)}.len != 0)')
+				enc.writeln('${indent}if (${prefix_enc}${op}${c_name(field.name)}.len != 0)')
 			} else {
-				enc.writeln('${indent}if (val${op}${c_name(field.name)} != ${g.type_default(field.typ)})')
+				enc.writeln('${indent}if (${prefix_enc}${op}${c_name(field.name)} != ${g.type_default(field.typ)})')
 			}
 		}
 		if !is_js_prim(field_type) {

